akka-docs-dev/rst/java/http/routing-dsl/marshalling.rst

```diff
@@ -3,4 +3,41 @@
 Marshalling & Unmarshalling
 ===========================
 
-TODO
\ No newline at end of file
+"Marshalling" is the process of converting a higher-level (object) structure into some kind of lower-level
+representation (and vice versa), often a binary wire format. Other popular names for it are "Serialization" or
+"Pickling".
+
+In akka-http "Marshalling" means the conversion of an object of type T into an HttpEntity, which forms the entity body
+of an HTTP request or response (depending on whether used on the client or server side).
+
+Marshalling
+-----------
+
+On the server-side marshalling is used to convert a application-domain object to a response (entity). Requests can
+contain an ``Accept`` header that lists acceptable content types for the client. A marshaller contains the logic to
+negotiate the result content types based on the ``Accept`` and the ``AcceptCharset`` headers.
+
+Marshallers can be specified when completing a request with ``RequestContext.completeAs`` or by using the ``BasicDirectives.completeAs``
+directives.
+
+These marshallers are provided by akka-http:
```

Comment:
Minor:  We should consolidate up how we write akka-http or Akka Http or Akka HTTP at some point.